protostar/cairo_testing/cairo_test_cheatcode_factory.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, cast
+
+from protostar.cheatable_starknet.cheatable_cached_state import (
+    CheatableCachedState,
+)
+from protostar.cheatable_starknet.cheaters.block_info import BlockInfoCairoCheater
+from protostar.cheatable_starknet.cheaters.contracts import ContractsCairoCheater
+
+from protostar.cheatable_starknet.cheaters import (
+    CairoCheaters,
+)
+from protostar.cairo_testing.cairo_test_execution_state import (
+    CairoTestExecutionState,
+)
+from protostar.cheatable_starknet.cheatcodes.cairo_cheatcode import CairoCheatcode
+from protostar.cheatable_starknet.cheatcodes.declare_cairo_cheatcode import (
+    DeclareCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.deploy_cairo_cheatcode import (
+    DeployCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.deploy_contract_cairo_cheatcode import (
+    DeployContractCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.invoke_cairo_cheatcode import (
+    InvokeCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.prepare_cairo_cheatcode import (
+    PrepareCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.roll_cairo_cheatcode import (
+    RollCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.warp_cairo_cheatcode import (
+    WarpCairoCheatcode,
+)
+from protostar.cheatable_starknet.cheatcodes.call_cairo_cheatcode import (
+    CallCairoCheatcode,
+)
+
+
+class CairoTestCheatcodeFactory:
+    def __init__(
+        self,
+        state: CairoTestExecutionState,
```

Comment:
```suggestion
        cheatable_cached_state: CheatableCachedState,
```

Reply:
+1, but I think it should be adressed in a separate PR, alongside with that we should stop using `Starknet` at all, from variouse preformance reasons